x/btcstaking/keeper/msg_server.go

```diff
@@ -160,275 +163,148 @@ func caluculateMinimumUnbondingValue(
 	return minUnbondingOutputValue
 }
 
-// CreateBTCDelegation creates a BTC delegation
-// TODO: refactor this handler. It's now too convoluted
-func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCreateBTCDelegation) (*types.MsgCreateBTCDelegationResponse, error) {
-	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), types.MetricsKeyCreateBTCDelegation)
-
-	ctx := sdk.UnwrapSDKContext(goCtx)
-	// basic stateless checks
-	if err := req.ValidateBasic(); err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "%v", err)
-	}
-
-	vp := ms.GetParamsWithVersion(ctx)
-	btccParams := ms.btccKeeper.GetParams(ctx)
-	kValue, wValue := btccParams.BtcConfirmationDepth, btccParams.CheckpointFinalizationTimeout
-
-	minUnbondingTime := types.MinimumUnbondingTime(vp.Params, btccParams)
+type ParamsValidationResult struct {
+	StakingOutputIdx   uint32
+	UnbondingOutputIdx uint32
+}
 
+// ValidateParams validates parsed message against parameters
+func ValidateParams(
```

Comment:
What do you think we move the params-related code to a separate file or a different package?

And would it be better if we make `ValidateParams` a method of `ParsedCreateDelegationMessage`?

Reply:
hmm I guess it can be moved to types as it really do not touch/modify the state.

As for making it a method on `ParsedCreateDelegationMessage`, I am slightly against mostly due to the fact validating against the parameters is not really a property of `ParsedCreateDelegationMessage` (I know this sounds weird 😅 ) but rather operation that is by using `ParsedCreateDelegationMessage` with other pieces of data.  

Maybe in other words, `ParsedCreateDelegationMessage` is just data transfer object without any assumptions how it will be used. Other functions, object by use it for its own purposes, but the dto object should be as bare as possible.